lib/cocina/models/validators/language_tag_validator.rb

```diff
@@ -17,31 +17,67 @@ def initialize(clazz, attributes)
         def validate
           return unless meets_preconditions?
 
-          return if valid_language_tag?
+          return if invalid_files.empty?
 
-          raise ValidationError, 'The provided language tag is not valid according to RFC 4646: ' \
-                                 "#{attributes[:languageTag]}"
+          raise ValidationError, 'Some files have invalid language tags according to RFC 4646: ' \
+                                 "#{invalid_filenames_with_language_tags.join(', ')}"
         end
 
         private
 
         attr_reader :clazz, :attributes
 
         def meets_preconditions?
-          file? && attributes[:languageTag].present?
+          dro?
         end
 
-        def file?
-          (clazz::TYPES & File::TYPES).any?
+        def dro?
+          (clazz::TYPES & DRO::TYPES).any?
         rescue NameError
           false
         end
 
-        def valid_language_tag?
-          parsed_tag = I18n::Locale::Tag::Rfc4646.tag(attributes[:languageTag])
+        def valid_language_tag?(file)
+          parsed_tag = I18n::Locale::Tag::Rfc4646.tag(file[:languageTag])
 
           parsed_tag.present? && parsed_tag.is_a?(I18n::Locale::Tag::Rfc4646)
         end
+
+        def invalid_files
+          @invalid_files ||=
+            [].tap do |invalid_files|
+              language_tag_files.each do |file|
```

Comment:
Here and below. This seems more idiomatic than tap:
```
invalid_files = language_tag_files.reject {|file| valid_language_tag?(file) }
```

Reply:
ah yeah, i like that, i cribbed the current implementation off `DarkValidator`, and had meant to circle back to do something more concise/idiomatic, but obviously forgot some cleanup between this and the straggler `puts`.